### DISCUSSION TITLE
How to add locale support from laravel-localization?

### DISCUSSION THREAD

**Discussion by AndriiTereshchuk** [tighten/ziggy#608] [Q&A]
I use this extension https://github.com/mcamara/laravel-localization, there are middlewares, but they are not displayed in the url

**Comment by robertosanval** [comment]
Duplicated of #425

**Comment by galexpert** [comment]
Hello, help me to set the default locale , but should not be displayed in the url
i also use this extension https://github.com/mcamara/laravel-localization, there are middlewares,
here is my router
Route::group([
/* 'prefix' => '{locale?}',*/
'middleware' => [ 'setLocale' ]
], function()
{
Route::get('{locale?}/real-estate/{slug}', [Modules\Realty\Http\Controllers\MainController::class, 'show'])->name('realtyvue.item');
});

here is my middleware setLocale

public function handle(Request $request, Closure $next)
{
URL::defaults(['locale' => LaravelLocalization::setLocale()]); // en, uk, pl
return $next($request);
}

when local == uk
http://domain.loc/uk/real-estate/17-shop Great result!!!!  200 ok

when local == pl
http://domain.loc/pl/real-estate/17-shop Great result!!!! 200 ok

when local == en
I need the default language not to be displayed in the url
in middleware
URL::defaults(['locale' => null]);

in vue
http://domain.loc//real-estate/17-shop Wrong bad result!!!! 404 error

and the result should be
http://domain.loc/real-estate/17-shop Great result!!!!  200 ok

Please help me to set up ziggy or laravel correctly when 'locale' must be null

**Comment by ctf0** [comment]
the [Default parameter values](https://github.com/tighten/ziggy?tab=readme-ov-file#default-parameter-values) doesnt work, you will need to share the locale with inertia from the mw, then use it with each route, i couldnt find away to tell ziggy to use the locale as a default url prop globally

**Reply by thaqebon** [reply]
> the [Default parameter values](https://github.com/tighten/ziggy?tab=readme-ov-file#default-parameter-values) doesnt work, you will need to share the locale with inertia from the mw, then use it with each route, i couldnt find away to tell ziggy to use the locale as a default url prop globally

I’ve submitted a PR to address this issue.

In the meantime, you can create your own wrapper around the `route()` function that retrieves the default locale and passes it to Ziggy’s `route()` helper.
